### PR TITLE
[FSDE-1336] Add support for checkbox indeterminate state

### PR DESCRIPTION
**Pull request by josecarlosgonzalezv**
- [x] Chi documentation
- [x] Chi tests

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-141/1/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://539a1804d4bf53d68b23b5b70904537d0eba6bac-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-141/2/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://65e82616af582df398cc816d453cbf94876c1414-chi-pr.enterprise.ctl.io

File: src/website/views/components/forms/checkbox/_examples.pug [hunk 1]
```diff
@@ -49,6 +49,42 @@ p.-text
         </div>
       </div>
 
+h3 Indeterminate
+p.-text
+  | Use the <code>indeterminate</code> property to set the default value of
```
Comment (janibolkvadze) [inline src/website/views/components/forms/checkbox/_examples.pug:54]:
```suggestion
  | A checkbox cannot be set to indeterminate state by an HTML attribute - it must be set by JavaScript.
```

File: src/website/views/components/forms/checkbox/_examples.pug [hunk 2]
```diff
@@ -49,6 +49,42 @@ p.-text
...
+  | a checkbox to indeterminate and then set the class <code>-indeterminate</code> to style.
```
Comment (janibolkvadze) [inline src/website/views/components/forms/checkbox/_examples.pug:55]:
```suggestion
  | Use the class <code>-indeterminate</code> to achieve the corresponding style of the checkbox.
```

File: src/website/views/components/forms/checkbox/_examples.pug [hunk 3]
```diff
@@ -49,6 +49,42 @@ p.-text
...
+
+.example.-mb--3
+  .-p--3
+    .chi-form__item
+      .chi-checkbox
+        input(type="checkbox" class="chi-checkbox__input" id="checkbox-ind1")
```
Comment (janibolkvadze) [inline src/website/views/components/forms/checkbox/_examples.pug:61]:
```suggestion
        input(type="checkbox" class="chi-checkbox__input -indeterminate" id="checkbox-ind1")
```

File: src/website/views/components/forms/checkbox/_examples.pug [hunk 4]
```diff
@@ -49,6 +49,42 @@ p.-text
...
+          <label class="chi-checkbox__label" for="checkbox-ind1">Checkbox</label>
+        </div>
+      </div>
+
+      <!-- JavaScript -->
+      <script>
+        var checkbox = document.getElementById('checkbox-ind1');
+        
+        checkbox.indeterminate = true;
+        checkbox.classList.add("-indeterminate");
```
Comment (janibolkvadze) [inline src/website/views/components/forms/checkbox/_examples.pug:85]:
```suggestion
```

File: src/website/views/components/forms/checkbox/_examples.pug [hunk 5]
```diff
@@ -505,5 +541,10 @@ script.
     function() {
       chi.tab(document.querySelectorAll('.chi-tabs-panel .chi-tabs'));
       chi.popover(document.getElementById('example__help-button'));
+
+      var checkbox = document.getElementById("checkbox-ind1");
+      
+      checkbox.indeterminate = true;
+      checkbox.classList.add("-indeterminate");
```
Comment (janibolkvadze) [inline src/website/views/components/forms/checkbox/_examples.pug:548]:
```suggestion
```

File: src/website/views/components/forms/checkbox/_properties.pug [hunk 6]
```diff
@@ -1,6 +1,29 @@
 h2 Web Component
 h3 Properties
-.chi-card.-align--center.-empty
```
Comment (janibolkvadze) [inline src/website/views/components/forms/checkbox/_properties.pug:3]: Please, revert the change as JS property indeterminate is generic, not Chi specific.

File: test/chi/components/checkbox.pug [hunk 7]
```diff
@@ -20,19 +20,33 @@ title: Checkbox
...
             disabled=state == 'disabled' ? 'disabled' : false
             )
             label.chi-checkbox__label(for=`checkbox--${element}`)= `Checkbox-${element}`
+
+script.
+  document.addEventListener(
+    'DOMContentLoaded',
+    function() {
+      var checkbox = document.getElementById("checkbox-4");
+      var checkboxList = ["checkbox--1--indeterminate", "checkbox--2--indeterminate", "checkbox--3--indeterminate", "checkbox--4--indeterminate", "checkbox--5--indeterminate"];
```
Comment (janibolkvadze) [inline test/chi/components/checkbox.pug:45]:
```suggestion
      var checkboxList = ["checkbox-4", "checkbox--1--indeterminate", "checkbox--2--indeterminate", "checkbox--3--indeterminate", "checkbox--4--indeterminate", "checkbox--5--indeterminate"];
```

File: test/chi/components/checkbox.pug [hunk 8]
```diff
@@ -20,19 +20,33 @@ title: Checkbox
...
+
+      checkbox.indeterminate = true;
```
Comment (janibolkvadze) [inline test/chi/components/checkbox.pug:47]:
```suggestion
```

**Review by janibolkvadze — CHANGES_REQUESTED** [review]

File: test/chi/components/checkbox.pug (hunk 8 above)
Comment (josecarlosgonzalezv) [reply test/chi/components/checkbox.pug]: Done.

File: test/chi/components/checkbox.pug (hunk 7 above)
Comment (josecarlosgonzalezv) [reply test/chi/components/checkbox.pug]: Done.

File: src/website/views/components/forms/checkbox/_properties.pug (hunk 6 above)
Comment (josecarlosgonzalezv) [reply src/website/views/components/forms/checkbox/_properties.pug]: Done.

File: src/website/views/components/forms/checkbox/_examples.pug (hunk 4 above)
Comment (josecarlosgonzalezv) [reply src/website/views/components/forms/checkbox/_examples.pug]: Done.

File: src/website/views/components/forms/checkbox/_examples.pug (hunk 5 above)
Comment (josecarlosgonzalezv) [reply src/website/views/components/forms/checkbox/_examples.pug]: Done.

File: src/website/views/components/forms/checkbox/_examples.pug (hunk 3 above)
Comment (josecarlosgonzalezv) [reply src/website/views/components/forms/checkbox/_examples.pug]: Done.

File: src/website/views/components/forms/checkbox/_examples.pug (hunk 2 above)
Comment (josecarlosgonzalezv) [reply src/website/views/components/forms/checkbox/_examples.pug]: Done.

File: src/website/views/components/forms/checkbox/_examples.pug (hunk 1 above)
Comment (josecarlosgonzalezv) [reply src/website/views/components/forms/checkbox/_examples.pug]: Done.

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-141/3/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://f78c587cbc6d3ea3cda9f2b50fea3c48db5048fc-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-141/4/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://f78c587cbc6d3ea3cda9f2b50fea3c48db5048fc-chi-pr.enterprise.ctl.io

**Review by janibolkvadze — APPROVED** [review]

**Comment by janibolkvadze** [comment]
@evansito I'm unable to mark you as a reviewer. Please, review the PR.

**Review by mattnickles — APPROVED** [review]

**Review by evansito — APPROVED** [review]
